halo/evmstaking2/keeper/keeper_internal_test.go

```diff
@@ -41,23 +41,50 @@ func TestInsertEVMEvents(t *testing.T) {
 		},
 	}
 
-	keeper, sdkCtx := setupKeeper(t)
+	submissionDelay := int64(5)
+
+	keeper, ctx := setupKeeper(t, submissionDelay)
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			sdkCtx = sdkCtx.WithBlockHeight(test.height)
+			ctx = ctx.WithBlockHeight(test.height)
 
-			err := keeper.Deliver(sdkCtx, common.Hash{}, test.event)
+			err := keeper.Deliver(ctx, common.Hash{}, test.event)
 			require.NoError(t, err)
 
-			event, err := keeper.eventsTable.Get(sdkCtx, test.insertedID)
+			event, err := keeper.eventsTable.Get(ctx, test.insertedID)
 			require.NoError(t, err)
 			require.Equal(t, test.event.Address, event.GetEvent().Address)
 		})
 	}
+
+	// Make sure no submission happens for heights in the range 2 to 4
+	for h := int64(2); h <= 4; h++ {
```

Comment:
nit: less magic numbers makes things slightly easier to read, can do `h < submissionDelay` for example